qcodes/instrument/channel.py

```diff
@@ -216,46 +228,54 @@ def __init__(self, parent: InstrumentBase,
         # channels. This will eventually become a locked tuple.
         self._channels: Sequence[InstrumentChannel]
         if chan_list is None:
-            self._locked = False
-            self._channels = []
+            self._channels = ()
```

Comment:
as discussed offline, let's make this type list which will make the code a bit simpler, especially in the channellist subclass